### ISSUE TITLE
Error when updating a feature in gdb file.

### ISSUE THREAD

**Issue by InsolublePancake** [OSGeo/gdal#7504]
## Expected behavior and actual behavior.

Expected behaviour: update the validated geometry in a gdb file.
Actual behaviour: RuntimeError: Error occurred in /src/gdal-3.6.2/ogr/ogrsf_frmts/openfilegdb/filegdbtable.cpp at line 1516

## Steps to reproduce the problem.

```
in_path = 'swse.gdb'

ds = gdal.OpenEx(in_path, gdal.GA_Update, open_options=[], allowed_drivers=[])
lyr = ds.GetLayer()
lyr.ResetReading()
for ftr in lyr:    
    geom = ftr.geometry()
    
    if not geom.IsValid():
        geom = geom.MakeValid()

    ftr.SetGeometry(geom)
    lyr.SetFeature(ftr)
    lyr.SyncToDisk()
```

## Operating system

Rocky Linux version 8.7

## GDAL version and provenance

3.6.2 install into a Python 3.8.13 venv using pip install gdal=3.6.2

**Comment by jratike80** [comment]
Could you capture the invalid geometry and how MakeValid corrects it as WKT?

**Comment by InsolublePancake** [comment]
Unfortunately this hasn't solved my problem. I think there may be problem that isn't geometry related. If I modify the above code so that instead of validating the geometry it sets each geometry to a simple polygon, I get a similar error:

```
ds = gdal.OpenEx(in_path, gdal.GA_Update, open_options=[], allowed_drivers=[])
lyr = ds.GetLayer()
lyr.ResetReading()
for ftr in lyr:    
    geom = ftr.geometry()
    geom = ogr.CreateGeometryFromWkt('POLYGON ((30 10, 10 20, 20 40, 40 40, 30 10))')
       
    ftr.SetGeometry(geom)
    lyr.SetFeature(ftr)
    lyr.SyncToDisk()

RuntimeError: Error occurred in /src/gdal-3.6.2/ogr/ogrsf_frmts/openfilegdb/filegdbtable.cpp at line 1516
```

**Comment by InsolublePancake** [comment]
@rouault thanks for the insight into make_valid. I will add this to my code elsewhere.

**Comment by InsolublePancake** [comment]
@jratike80 I can't see anything particularly wrong with any of the geometries. They are mostly multipolygons, but without geometry errors that I can see.

**Comment by jratike80** [comment]
But if they are all valid, then your code should not try to do anything, right? `if not geom.IsValid():`

**Comment by InsolublePancake** [comment]
> But if they are all valid, then your code should not try to do anything, right? `if not geom.IsValid():`

Actually, I get the same error whether I run the make_valid or not, and also if I replace each geometry with an simply polygon.

**Comment by jratike80** [comment]
All features in the table seem to be multipolygons, even the single-parted. Have you tried
`geom = ogr.CreateGeometryFromWkt('MULTIPOLYGON (((30 10, 10 20, 20 40, 40 40, 30 10)))')

I tried to do something similar with ogr2ogr but I did also get an error around the same place
`ERROR 1: Error occurred in E:\buildsystem\src\gdal-vc17x64\ogr\ogrsf_frmts\openfilegdb\filegdbtable.cpp at line 1518`.
It is possible that my test was not good, I have no experience on file gdb.

**Comment by InsolublePancake** [comment]
I've just tried your suggestion, but I get the same error.

**Comment by InsolublePancake** [comment]
@rouault, yes the file was created using GDAL. The company where I work has built up a private library that uses GDAL. We've recently updated the GDAL version made adjustments to swap the FileGDB driver for OpenFileGDB. This error has occurred in a system test. The test uses a lot of inhouse functions to create a raster from an array then polygonise it into a vector file - most of this goes on in-memory so it's difficult to reproduce in a succinct way.

**Comment by InsolublePancake** [comment]
Ok, thanks I'll see what I can do.

**Comment by InsolublePancake** [comment]
Ok, I have some code that replicates the issue.
I've attached a file with an array in the form of a list. I hope this gives you what you need.
```
[rows.txt](https://github.com/OSGeo/gdal/files/11080231/rows.txt)
arr_band = np.array(rows) #rows is a list in the attached file.
temp_dir = '.'
file_name = "test.gdb"
driverName = "OpenFileGDB"
drv = gdal.GetDriverByName(driverName)
ds = drv.Create(os.path.join(temp_dir, file_name), 0, 0, 0, gdal.GDT_Unknown )
lyr = ds.CreateLayer("test", None, ogr.wkbPolygon)

fields = [('Id', 0), ('a', 0), ('b', 4), ('Shape_Length', 2), ('Shape_Area', 2)]
for field in fields:
    lyr.CreateField(ogr.FieldDefn(*field))
        
pixel_value_field = 1
nodata = 15
options = ["8CONNECTED=8"]
dtype = 5
# create in_memory vector layer
ysize, xsize = arr_band.shape
drv = gdal.GetDriverByName("MEM")
raster_block = drv.Create("temp", xsize, ysize, 1, dtype)
raster_block.SetProjection(wkt_srs)
raster_block.SetGeoTransform(gt)
bnd = raster_block.GetRasterBand(1)

bnd.SetNoDataValue(nodata)
bnd.WriteArray(arr_band)

gdal.Polygonize(bnd, bnd.GetMaskBand(), lyr, pixel_value_field, options)

lyr.ResetReading()

for ftr in lyr:
    geom = ftr.geometry()
    if not geom.IsValid():
        new_geom = geom.MakeValid().RemoveLowerDimensionSubGeoms()
               
        ftr.SetGeometry(new_geom)
        lyr.SetFeature(ftr)
    lyr.SyncToDisk()

ds = None
```

**Comment by rouault** [comment]
Fix in https://github.com/OSGeo/gdal/pull/7525.
@InsolublePancake Thanks a lot for taking time to come up with a reproducer and helping spotting this annoying bug!
